ASTProcessor/src/main/java/edu/usb/argos/ASTProcessor/FileReader/JavaParserFileReaderByText.java

```diff
@@ -0,0 +1,42 @@
+package edu.usb.argos.ASTProcessor;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.TypeDeclaration;
+
+
+public class JavaParserFileReaderByText implements IFileAnalyzer<CompilationUnit, String> {
+    @Override
+    public CompilationUnit read(String text) {
+        JavaParser javaParser = new JavaParser(new ParserConfiguration());
+        ParseResult<CompilationUnit> parseResult = javaParser.parse(text);
+        if (parseResult.isSuccessful() && parseResult.getResult().isPresent()) {
+            CompilationUnit compilationUnit = parseResult.getResult().get();
+            return compilationUnit;
+        } else {
+            throw new RuntimeException("Error to analyze de code");
+        }
+    }
+
+    public void validateInput(String text) {
```

Comment:
could you make a class to handle errors or validations and avoid having multiple responsibilities in one class, please?

Reply:
Done